consensus/core/src/core.rs

```diff
@@ -487,31 +472,77 @@ impl Core {
                         .join(",")
                 );
             }
+            self.commit_observer.handle_commit(committed_leaders)
+        } else {
+            let mut committed_subdags = Vec::new();
+            // TODO: Add optimization to abort early without quorum for a round.
+            loop {
+                // LeaderSchedule has a limit to how many sequenced leaders can be committed
+                // before a change is triggered. Calling into leader schedule will get you
+                // how many commits till next leader change. We will loop back and recalculate
+                // any discarded leaders with the new schedule.
+                let sequenced_leaders = self
+                    .committer
+                    .try_commit(self.last_decided_leader)
+                    .into_iter()
+                    .take(
+                        self.leader_schedule
+                            .commits_until_leader_schedule_update(self.dag_state.clone()),
+                    )
+                    .collect::<Vec<_>>();
+
+                if sequenced_leaders.is_empty() {
+                    break;
+                }
```

Comment:
```suggestion
                let Some(last) = sequenced_leaders.last() else {
                    break;
                };
```
and we can remove the `if let Some(last) = sequenced_leaders.last()` and `if !committed_leaders.is_empty()` conditions below.